### ISSUE TITLE
Show @since and @deprecated in `h` in IEx

### ISSUE THREAD

**Issue by josevalim** [elixir-lang/elixir#7217]
Depends on #7198.

**Comment by lackac** [comment]
I have started working on this. Any preference on how to display metadata? I thought of displaying it in alphabetical order right below typespecs, indented and tabulated:

```
iex(5)> h Enum.chunk_every/4

            def chunk_every(enumerable, count, step, leftover \\ [])

    @spec chunk_every(t(), pos_integer(), pos_integer(), t() | :discard) :: [list()]

    - since    "1.5.0"
    - defaults 1

Returns list of lists containing count items each, where each new chunk starts
step elements into the enumerable.
...
```

I included in the example above, but I don't think we should display `defaults` as it doesn't provide additional information and its use is internal.

Another question that comes to mind is whether to display arbitrary metadata (I think yes) or just show `since` and `deprecated`?

**Comment by fertapric** [comment]
What do you think about?

![screen shot 2018-07-11 at 11 48 16 am](https://user-images.githubusercontent.com/651203/42564353-d2908da4-8500-11e8-8420-87ecb6bb9538.png)

**Comment by lackac** [comment]
Looks good. I appreciate the extra effort of rendering it. Do you think it
needs a heading and or indentation?

**Comment by josevalim** [comment]
I think we need to consider if we want to print all metadata first. For example, authors could be a list: how are we going to detect and show it? are we simply going to use inspect? if so, then it would rather look like this:

```elixir
since: "1.5.0"
deprecated: "Use foo.bar instead"
authors: ["Jane", "Mary"]
```

Which is much more verbose. My suggestion therefore is to go with @fertapric's styling but only print `since` and `deprecated` for now. We can add more in the future accordingly.

**Comment by fertapric** [comment]
> Do you think it needs a heading and or indentation?

I wouldn't add a heading. Regarding indentation, I think it's enough with the colors :)

> are we simply going to use inspect?

That's what I thought when I was doing the mockup 😅 

But 👌 with doing small steps and seeing how it evolves.

**Comment by josevalim** [comment]
Alright, once this is ready we should be good to ship the first release candidate for v1.7. :D

**Comment by josevalim** [comment]
Closing in favor of #7886.
